sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationRefreshUtil.java

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.implementation;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.data.appconfiguration.models.ConfigurationSetting;
+import com.azure.data.appconfiguration.models.SettingSelector;
+import com.azure.spring.cloud.config.pipline.policies.BaseAppConfigurationPolicy;
+import com.azure.spring.cloud.config.properties.AppConfigurationProviderProperties;
+import com.azure.spring.cloud.config.properties.AppConfigurationStoreMonitoring;
+import com.azure.spring.cloud.config.properties.FeatureFlagStore;
+
+class AppConfigurationRefreshUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppConfigurationPullRefresh.class);
+
+    /**
+     * Goes through each config store and checks if any of its keys need to be refreshed. If any store has a value that
+     * needs to be updated a refresh event is called after every store is checked.
+     *
+     * @return If a refresh event is called.
+     */
+    static RefreshEventData refreshStoresCheck(AppConfigurationProviderProperties appProperties,
```

Comment:
The only usage for the passed in `appProperties` is to construct the backoff, why don't we just pass in a Backoff object into this method?